homeassistant/components/homekit_controller/__init__.py

```diff
@@ -125,6 +113,8 @@ def _setup_characteristic(self, char):
             # pylint: disable=not-callable
             update_fn(result['value'])
 
+        self.async_schedule_update_ha_state()
```

Comment:
Is this called from async? because you're calling it from a sync method? 

If it's called from async and you don't need any work done in `async_update`, you could also use `self.async_write_ha_state()`

Reply:
It's non blocking code thats called from an async method elsewhere. Should i call it async_handle_new_values to make thise clear?

I will keep this in mind as I implement @MartinHjelmare suggestions.

Reply:
Have switched to `async_write_ha_state` and its now called from an explicitly async method.